packages/venia-ui/lib/components/CartPage/PriceAdjustments/CouponCode/couponCode.js

```diff
@@ -0,0 +1,211 @@
+import React, { useCallback, useEffect } from 'react';
+import { useLazyQuery, useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+import { PriceSummaryQuery } from '../../PriceSummary';
+import Button from '../../../Button';
+import { mergeClasses } from '../../../../classify';
+import defaultClasses from './couponCode.css';
+import { Form } from 'informed';
+import Field from '../../../Field';
+import TextInput from '../../../TextInput';
+
+export const APPLY_COUPON_MUTATION = gql`
+    mutation applyCouponToCart($cartId: String!, $couponCode: String!) {
+        applyCouponToCart(
+            input: { cart_id: $cartId, coupon_code: $couponCode }
+        ) {
+            cart {
+                id
+                # TODO: refetch discounts query
```

Comment:
Instead of `refetch` can we just query for data using fragments? Should we do that?

Reply:
@tjwiebell informed me that in order for us to be able to just fetch this data in the original query , and have it updated elsewhere such as price summary or applied coupons, we have to use fragments. Something for future me to look into :D